### PR TITLE
Fix rel-eng changelog for container images

### PR DESCRIPTION
**Pull request by ncounter**
## What does this PR change?

$subject

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: rel-eng stuff

- [x] **DONE**

## Test coverage
- No tests: rel-eng stuff

- [x] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by juliogonzalez — COMMENTED** [review]
Did you check this against a test project, so see what whill happen on the next submission after the changelog entry is added?

**Comment by ncounter** [comment]
> Did you check this against a test project, so see what whill happen on the next submission after the changelog entry is added?

Yes I extracted the slice of code and tested on the local project folder, and this is the behavior I see is the one I expect:
- If there is no changelog entry with the _version string_, it adds one.
- If there is one with the _version string_, it compares the value, if it is different it adds one.
- If there are multiple changelog with the _version string_, it takes only the first one (first at the top = latest) and it does the comparison as above
all that said compared to the version in `/web/conf/rhn_web.conf`

File: rel-eng/push-packages-to-obs.sh [hunk 1]
```diff
@@ -271,11 +271,12 @@ while read PKG_NAME; do
       PRODUCT_VERSION=$(echo ${PRODUCT_VERSION,,} | sed -r 's/ /-/g')
       sed "s/%PKG_VERSION%/${PRODUCT_VERSION}/g" -i $SRPM_PKG_DIR/Dockerfile
 
-      # if obs vs local version is different, add changelog about the bumping version
-      OSC_PKG_VERSION=$(${OSC} cat ${OBS_PROJ}/${PKG_NAME}/Dockerfile | grep '^LABEL org.opencontainers.image.version=')
-      OSC_PKG_VERSION=$(echo ${OSC_PKG_VERSION} | sed -n 's/.*=\"\(.*\)\"/\1/p')
-      if [ "${OSC_PKG_VERSION}" != "${PRODUCT_VERSION}" ]; then
-        ${OSC} vc ${SRPM_PKG_DIR} -m "Bump version to ${PRODUCT_VERSION}"
+      # add a changelog entry if it is not yet up to the latest version
```
Comment (juliogonzalez) [inline rel-eng/push-packages-to-obs.sh:274]: One minor suggestion, according to what we discussed 
```suggestion
      # add a changelog entry if it is not yet up to the latest version
      CHANGELOG_TEMPLATE="Update the image version to"
```

File: rel-eng/push-packages-to-obs.sh [hunk 2]
```diff
@@ -271,11 +271,12 @@ while read PKG_NAME; do
...
+      CHANGELOG_PKG_VERSION=$(cat ${SRPM_PKG_DIR}/${PKG_NAME}.changes | grep -m 1 "Update the image version to ")
+      CHANGELOG_PKG_VERSION=$(echo ${CHANGELOG_PKG_VERSION} | sed -n 's/.*Update\sthe\simage\sversion\sto\s\(.*\)$/\1/p')
```
Comment (juliogonzalez) [inline rel-eng/push-packages-to-obs.sh:276]:
```suggestion
      CHANGELOG_PKG_VERSION=$(echo ${CHANGELOG_PKG_VERSION} | sed -n "s/.* ${CHANGELOG_TEMPLATE} (.*\)$/\1/p")
```

File: rel-eng/push-packages-to-obs.sh [hunk 3]
```diff
@@ -271,11 +271,12 @@ while read PKG_NAME; do
...
+      if [ "${CHANGELOG_PKG_VERSION}" != "${PRODUCT_VERSION}" ]; then
+        echo "Adding a changelog entry for ${SRPM_PKG_DIR}"
+        ${OSC} vc ${SRPM_PKG_DIR} -m "Update the image version to ${PRODUCT_VERSION}"
```
Comment (juliogonzalez) [inline rel-eng/push-packages-to-obs.sh:279]:
```suggestion
        ${OSC} vc ${SRPM_PKG_DIR} -m "${CHANGELOG_TEMPLATE} ${PRODUCT_VERSION}"
```

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
One minor suggestion, abut what we discussed on Friday.

Make sure you retest, just in case! :-D

**Comment by ncounter** [comment]
Closing because of https://github.com/uyuni-project/uyuni/pull/5401

**Pull request closed without merging**